File: backend/HarborOfHope.API/Data/Entities/Resident.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HarborOfHope.API.Data.Entities;

[Table("residents")]
public class Resident
{
    [Key]
    [Column("resident_id")]
    public int ResidentId { get; set; }

    [Column("case_control_no")]
    public string? CaseControlNo { get; set; }

    [Column("internal_code")]
    public string? InternalCode { get; set; }

    [Column("safehouse_id")]
    public int SafehouseId { get; set; }

    [Column("case_status")]
    public string? CaseStatus { get; set; }

    [Column("sex")]
    public string? Sex { get; set; }

    [Column("date_of_birth")]
    public DateTime? DateOfBirth { get; set; }

    [Column("birth_status")]
    public string? BirthStatus { get; set; }

    [Column("place_of_birth")]
    public string? PlaceOfBirth { get; set; }

    [Column("religion")]
    public string? Religion { get; set; }

    [Column("case_category")]
    public string? CaseCategory { get; set; }

    [Column("sub_cat_orphaned")]
    public bool SubCatOrphaned { get; set; }

    [Column("sub_cat_trafficked")]
    public bool SubCatTrafficked { get; set; }

    [Column("sub_cat_child_labor")]
    public bool SubCatChildLabor { get; set; }

    [Column("sub_cat_physical_abuse")]
    public bool SubCatPhysicalAbuse { get; set; }

    [Column("sub_cat_sexual_abuse")]
    public bool SubCatSexualAbuse { get; set; }

    [Column("sub_cat_osaec")]
    public bool SubCatOsaec { get; set; }

    [Column("sub_cat_cicl")]
    public bool SubCatCicl { get; set; }

    [Column("sub_cat_at_risk")]
    public bool SubCatAtRisk { get; set; }

    [Column("sub_cat_street_child")]
    public bool SubCatStreetChild { get; set; }

    [Column("sub_cat_child_with_hiv")]
    public bool SubCatChildWithHiv { get; set; }

    [Column("is_pwd")]
    public bool IsPwd { get; set; }

    [Column("pwd_type")]
    public string? PwdType { get; set; }

    [Column("has_special_needs")]
    public bool HasSpecialNeeds { get; set; }

    [Column("special_needs_diagnosis")]
    public string? SpecialNeedsDiagnosis { get; set; }

    [Column("family_is_4ps")]
    public bool FamilyIs4ps { get; set; }

    [Column("family_solo_parent")]
    public bool FamilySoloParent { get; set; }

    [Column("family_indigenous")]
    public bool FamilyIndigenous { get; set; }

    [Column("family_parent_pwd")]
    public bool FamilyParentPwd { get; set; }

    [Column("family_informal_settler")]
    public bool FamilyInformalSettler { get; set; }

    [Column("date_of_admission")]
    public DateTime? DateOfAdmission { get; set; }

    [Column("age_upon_admission")]
    public string? AgeUponAdmission { get; set; }

    [Column("present_age")]
    public string? PresentAge { get; set; }

    [Column("length_of_stay")]
    public string? LengthOfStay { get; set; }

    [Column("referral_source")]
    public string? ReferralSource { get; set; }

    [Column("referring_agency_person")]
    public string? ReferringAgencyPerson { get; set; }

    [Column("date_colb_registered")]
    public DateTime? DateColbRegistered { get; set; }

    [Column("date_colb_obtained")]
    public DateTime? DateColbObtained { get; set; }

    [Column("assigned_social_worker")]
    public string? AssignedSocialWorker { get; set; }

    [Column("initial_case_assessment")]
    public string? InitialCaseAssessment { get; set; }

    [Column("date_case_study_prepared")]
    public DateTime? DateCaseStudyPrepared { get; set; }

    [Column("reintegration_type")]
    public string? ReintegrationType { get; set; }

    [Column("reintegration_status")]
    public string? ReintegrationStatus { get; set; }

    [Column("initial_risk_level")]
    public string? InitialRiskLevel { get; set; }

    [Column("current_risk_level")]
    public string? CurrentRiskLevel { get; set; }

    [Column("date_enrolled")]
    public DateTime? DateEnrolled { get; set; }

    [Column("date_closed")]
    public DateTime? DateClosed { get; set; }

    [Column("created_at")]
    public DateTime? CreatedAt { get; set; }

    [Column("notes_restricted")]
    public string? NotesRestricted { get; set; }

    // Navigation properties
    [ForeignKey("SafehouseId")]
    public Safehouse? Safehouse { get; set; }

    public ICollection<EducationRecord> EducationRecords { get; set; } = new List<EducationRecord>();
    public ICollection<HealthWellbeingRecord> HealthWellbeingRecords { get; set; } = new List<HealthWellbeingRecord>();
    public ICollection<ProcessRecording> ProcessRecordings { get; set; } = new List<ProcessRecording>();
    public ICollection<HomeVisitation> HomeVisitations { get; set; } = new List<HomeVisitation>();
    public ICollection<IncidentReport> IncidentReports { get; set; } = new List<IncidentReport>();
    public ICollection<InterventionPlan> InterventionPlans { get; set; } = new List<InterventionPlan>();
}

File: frontend/src/types/Resident.ts
export interface ResidentListItem {
  residentId: number;
  caseControlNo: string | null;
  internalCode: string | null;
  safehouseId: number;
  safehouseName: string | null;
  caseStatus: string | null;
  caseCategory: string | null;
  currentRiskLevel: string | null;
  dateOfAdmission: string | null;
  presentAge: string | null;
}

export interface ResidentDetail {
  residentId: number;
  caseControlNo: string | null;
  internalCode: string | null;
  safehouseId: number;
  safehouseName: string | null;
  caseStatus: string | null;
  sex: string | null;
  dateOfBirth: string | null;
  birthStatus: string | null;
  placeOfBirth: string | null;
  religion: string | null;
  caseCategory: string | null;

  // Sub-categories
  subCatOrphaned: boolean;
  subCatTrafficked: boolean;
  subCatChildLabor: boolean;
  subCatPhysicalAbuse: boolean;
  subCatSexualAbuse: boolean;
  subCatOsaec: boolean;
  subCatCicl: boolean;
  subCatAtRisk: boolean;
  subCatStreetChild: boolean;
  subCatChildWithHiv: boolean;

  // Disability and special needs
  isPwd: boolean;
  pwdType: string | null;
  hasSpecialNeeds: boolean;
  specialNeedsDiagnosis: string | null;

  // Family background
  familyIs4ps: boolean;
  familySoloParent: boolean;
  familyIndigenous: boolean;
  familyParentPwd: boolean;
  familyInformalSettler: boolean;

  // Dates and case details
  dateOfAdmission: string | null;
  ageUponAdmission: string | null;
  presentAge: string | null;
  lengthOfStay: string | null;
  referralSource: string | null;
  referringAgencyPerson: string | null;
  dateColbRegistered: string | null;
  dateColbObtained: string | null;
  assignedSocialWorker: string | null;
  initialCaseAssessment: string | null;
  dateCaseStudyPrepared: string | null;
  reintegrationType: string | null;
  reintegrationStatus: string | null;
  initialRiskLevel: string | null;
  currentRiskLevel: string | null;
  dateEnrolled: string | null;
  dateClosed: string | null;
  createdAt: string | null;
}

export type ResidentFormData = Omit<
  ResidentDetail,
  'residentId' | 'safehouseName' | 'createdAt'
>;

export interface ResidentQueryParams {
  page: number;
  pageSize: number;
  sortBy?: string;
  sortDir?: 'asc' | 'desc';
  search?: string;
  safehouseId?: number;
  status?: string;
  riskLevel?: string;
  category?: string;
}
